readthedocs/api/v3/views.py

```diff
@@ -0,0 +1,320 @@
+
+    Allowed via URL attributes:
+
+    * slug
+    * slug__contains
+    * name
+    * name__contains
+
+    ### Expandable fields
+
+    There are some fields that are not returned by default because they are
+    expensive to calculate. Although, they are available for those cases where
+    they are needed.
+
+    Allowed via ``?expand=`` URL attribue:
+
+    * users
+    * active_versions
+    * active_versions.last_build
+    * active_versions.last_build.confg
+
+
+    ### Examples:
+
+    * List my projects: ``/api/v3/projects/``
+    * List my projects with offset and limit: ``/api/v3/projects/?offset=10&limit=25``
+    * Filter list: ``/api/v3/projects/?name__contains=test``
+    * Retrieve only needed data: ``/api/v3/projects/?fields=slug,created``
+    * Retrieve specific project: ``/api/v3/projects/{project_slug}/``
+    * Expand required fields: ``/api/v3/projects/{project_slug}/?expand=active_versions``
+    * Translations of a project: ``/api/v3/projects/{project_slug}/translations/``
+    * Subprojects of a project: ``/api/v3/projects/{project_slug}/subprojects/``
+    * Superproject of a project: ``/api/v3/projects/{project_slug}/superproject/``
+
+    Go to [https://docs.readthedocs.io/en/stable/api/v3.html](https://docs.readthedocs.io/en/stable/api/v3.html)
+    for a complete documentation of the APIv3.
+    """  # noqa
+
+    model = Project
+    lookup_field = 'slug'
+    lookup_url_kwarg = 'project_slug'
+    serializer_class = ProjectSerializer
+    filterset_class = ProjectFilter
+    queryset = Project.objects.all()
+    permit_list_expands = [
+        'users',
+        'active_versions',
+        'active_versions.last_build',
+        'active_versions.last_build.config',
+    ]
+
+    def get_queryset(self):
+        # This could be a class attribute and managed on the ``APIAuthMixin`` in
+        # case we want to extend the ``prefetch_related`` to other views as
+        # well.
+        queryset = super().get_queryset()
+        return queryset.prefetch_related(
+            'related_projects',
+            'domains',
+            'tags',
+        )
+
+    def get_view_description(self, *args, **kwargs):  # pylint: disable=arguments-differ
+        """
+        Make valid links for the user's documentation browseable API.
+
+        If the user has already one project, we pick the first and make all the
+        links for that project. Otherwise, we default to the placeholder.
+        """
+        description = super().get_view_description(*args, **kwargs)
+
+        project = None
+        if self.request and self.request.user.is_authenticated():
+            project = self.request.user.projects.first()
+            if project:
+                # TODO: make the links clickable when ``kwargs.html=True``
+                return mark_safe(description.format(project_slug=project.slug))
+        return description
+
+    @action(detail=True, methods=['get'])
+    def superproject(self, request, project_slug):
```

Comment:
This is a good point.

When I wrote this I found that requesting the details of a project and having `subproject_of` and `translation_of` was more readable. On the other hand, if you have a and wants to know the `superproject` of it just hit this endpoint.

Now that you mention, it's true that it's the same but with different names depending on the context.

> Is this endpoint needed given that we have subproject_of?

If we are going to use the detailed serializer for that field as well, I don't think so. It would be only a "shortcut" which probably it's not useful. I can just remove it.